gt-scheduler/website
# Update Course Combination Algorithm

### Summary

Resolves #151



### Checklist

- [x] Generated combinations do not conflict with the events.
- [x] Update getCombinations's usage in CourseContainer.


### How to Test

- In `src/components/CombinationContainer/index.tsx`, assign a dummy events list of type `Immutable<Event[]>` to `events` after the `events` prop import from ScheduleContext
- Choose a term in the dev environment and add courses
- Experiment with different Event times by updating `events`, and check for conflicts in the Combinations tab (the combinations where sections conflict with the recurring event should not appear)
- An example `events`, which will reserve the time between 9:30am to 5:00pm on Monday and Wednesday.
```
events = castImmutable([
  {
    id: '123',
    name: 'test',
    period: {
      start: 570,
      end: 1020,
    },
    days: ['M', 'W'],
  },
]);
```
